test/functional/spam2_controller_test.rb

```diff
@@ -0,0 +1,5 @@
+require 'test_helper'
+
+class Spam2ControllerTest < ActionDispatch::IntegrationTest
+
```

Comment:
Before we merge this large pr, we need to implement testing for the different routes here. I think this is essential with functional tests for the batch actions and then possibly a system test for the dashboard.

Reply:
Yes, rightly said @Uzay-G . I was going to write this. Thanks for bringing the point.

Reply:
Sure, I am working on that. Thanks!!

Reply:
Functional tests are done, please review. Thanks!!